src/Analyzers/CSharp/Tests/UsePrimaryConstructor/UsePrimaryConstructorTests.cs

```diff
@@ -3291,4 +3291,296 @@ partial class C<T> : Base
+                """,
+            LanguageVersion = LanguageVersion.CSharp12,
+        }.RunAsync();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70868")]
+    public async Task TestSideEffects3_A()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    public int B { get; }
+                    public int A { get; }
+
+                    public C(int value)
+                    {
+                        A = ++value;
+                        B = ++value;
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersion.CSharp12,
+        }.RunAsync();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70868")]
+    public async Task TestSideEffects4()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                partial class C
+                {
+                    public int A;
+
+                    public C(int value)
+                    {
+                        A = ++value;
+                        B = ++value;
+                    }
+                }
+
+                partial class C
+                {
+                    public int B;
+                }
+                """,
+            LanguageVersion = LanguageVersion.CSharp12,
+        }.RunAsync();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70868")]
+    public async Task TestSideEffects5()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    public int A;
+                    public int B;
+
+                    public [|C|](int value1, int value2)
+                    {
+                        A = ++value1;
+                        B = ++value2;
+                    }
+                }
+                """,
+            FixedCode = """
+                class C(int value1, int value2)
+                {
+                    public int A = ++value1;
+                    public int B = ++value2;
+                }
+                """,
+            LanguageVersion = LanguageVersion.CSharp12,
+        }.RunAsync();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70868")]
+    public async Task TestSideEffects5_A()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    public int B;
+                    public int A;
+
+                    public [|C|](int value1, int value2)
+                    {
+                        A = ++value1;
+                        B = ++value2;
+                    }
+                }
+                """,
+            FixedCode = """
+                class C(int value1, int value2)
+                {
+                    public int B = ++value2;
+                    public int A = ++value1;
```

Comment:
What behavior would be expected in a case like the following?
```cs
class Pair
{
    public int First;
    public int Second;

    public Pair(Stack<int> items)
    {
        Second = items.Pop();
        First = items.Pop();
    }
}
```

Reply:
@RikkiGibson badness :)

Reply:
I didn't mean to imply the fixer behavior should be adjusted from what you already have based on the possibility of this scenario. More that I think it's good to have a couple tests here and there for scenarios which show where the "boundaries" of the fixer's heuristics are. However, I also don't see a need to go back and add the test now.